sdks/python/apache_beam/ml/inference/pytorch_inference.py

```diff
@@ -281,6 +293,18 @@ def run_inference(
         batched_tensors = _convert_to_device(batched_tensors, self._device)
         key_to_batched_tensors[key] = batched_tensors
       predictions = model(**key_to_batched_tensors, **inference_args)
+      if isinstance(predictions, dict):
+        # Go from one dictionary of type: {key_type1: Iterable<val_type1>,
+        # key_type2: Iterable<val_type2>, ...} where each Iterable is of
+        # length batch_size, to a list of dictionaries:
+        # [{key_type1: value_type1, key_type2: value_type2}]
+        predictions_per_tensor = [
+            dict(zip(predictions, v)) for v in zip(*predictions.values())
```

Comment:
nit: To be more explicit that we want the keys here
```suggestion
            dict(zip(predictions.keys(), v)) for v in zip(*predictions.values())
```